filematch/patternmatcher.go

```diff
@@ -0,0 +1,230 @@
+package filematch
```

Comment:
I didn't add `// import` comments here, so that we don't have to remove those when moving to tonistiigi/fsutil